libs/ui/layout/src/lib/chart/chart.component.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  ViewChild,
+} from '@angular/core'
+import {
+  Chart,
+  BarController,
+  BarElement,
+  CategoryScale,
+  LinearScale,
+  LineController,
+  LineElement,
+  PointElement,
+  PieController,
+  ArcElement,
+  ScatterController,
+  Tooltip,
+  Colors,
+  Legend,
+  ChartType,
+} from 'chart.js'
+import { InputChartType } from './chart.model'
+
+Chart.register(
+  BarController,
+  BarElement,
+  CategoryScale,
+  LinearScale,
+  LineController,
+  LineElement,
+  PointElement,
+  PieController,
+  ArcElement,
+  ScatterController,
+  Tooltip,
+  Colors,
+  Legend
+)
+
+@Component({
+  selector: 'gn-ui-chart',
+  templateUrl: './chart.component.html',
+  styleUrls: ['./chart.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ChartComponent implements AfterViewInit {
+  private xAxisValue: string
+  private yAxisValue: string
+  private chartTypeValue: InputChartType
+
+  chart: Chart
+
+  @Input() data: Array<unknown>
+  @Input() set xAxis(value: string) {
+    if (this.xAxisValue) {
+      this.xAxisValue = value
+      this.updateChartData()
+    } else {
+      this.xAxisValue = value
+    }
+  }
+  @Input() set yAxis(value: string) {
+    if (this.yAxisValue) {
```

Comment:
in what case this would be `null` ?
what happens if you call `updateChartData()` while it's null ?

To me those 2 setters are quite verbose.

Reply:
On chart creation, these properies are `undefined`, so the `if` is only fulfilled on updates. You're right, it's still quite verbose.